hack/compare-k8s-feature-gates.sh

```diff
@@ -0,0 +1,61 @@
+  echo "> compare-k8s-feature-gates.sh [ -h | <old version> <new version> ]"
+  echo
+  echo ">> For example: compare-k8s-feature-gates.sh 1.26 1.27"
+
+  exit 0
+}
+
+if [ "$1" == "-h" ] || [ "$#" -ne 2 ]; then
+  usage
+fi
+
+versions=("$1" "$2")
+out_dir=$(mktemp -d)
+function cleanup_output {
+    rm -rf "$out_dir"
+}
+trap cleanup_output EXIT
+
+for version in "${versions[@]}"; do
+  versions_dir=test/compatibility_lifecycle/reference
+  
+  # TODO Drop this when Kubernetes v1.34 gets released
+  if [ "$version" == "1.32" ]; then
+    versions_dir="test/featuregates_linter/test_data"
+  fi
+
+ # Get the list of versioned and unversioned feature gates. After v1.34+ unversioned_feature_list.yaml should most likely be dropped
+  wget -q -O - "https://raw.githubusercontent.com/kubernetes/kubernetes/release-${version}/${versions_dir}/versioned_feature_list.yaml" > "${out_dir}/versioned_featuregates_${version}.yaml"
+  wget -q -O - "https://raw.githubusercontent.com/kubernetes/kubernetes/release-${version}/${versions_dir}/unversioned_feature_list.yaml" > "${out_dir}/unversioned_featuregates_${version}.yaml"
+  # Extract only the names of the feature gates
+  yq '.[] | .name' "${out_dir}/versioned_featuregates_${version}.yaml" > "${out_dir}/featuregates_list_${version}.yaml"
+  yq '.[] | .name' "${out_dir}/unversioned_featuregates_${version}.yaml" >> "${out_dir}/featuregates_list_${version}.yaml"
+  # Sort feature gate list for the diff to function correctly
+  sort -o "${out_dir}/featuregates_list_${version}.yaml" "${out_dir}/featuregates_list_${version}.yaml"
+done
+
+echo "Feature gates added in $2 compared to $1:"
+diff "${out_dir}/featuregates_list_${1}.yaml" "${out_dir}/featuregates_list_${2}.yaml" | grep '>' | awk '{print $2}'
+echo
+echo "Feature gates removed in $2 compared to $1:"
+diff "${out_dir}/featuregates_list_${1}.yaml" "${out_dir}/featuregates_list_${2}.yaml" | grep '<' | awk '{print $2}'
+echo
+echo "Feature gates locked to default true:"
+# Get all feature gate names that have a version spec containing $2, are locked to default with default value of true
+yq '.[] | select(.versionedSpecs[] | select(.version == "'$2'" and .lockToDefault == true and .default == true)) | .name' "${out_dir}/versioned_featuregates_${version}.yaml"
+yq '.[] | select(.versionedSpecs[] | select(.version == "'$2'" and .lockToDefault == true and .default == true)) | .name' "${out_dir}/unversioned_featuregates_${version}.yaml"
+echo
+echo "Feature gates locked to default false:"
+# Get all feature gate names that have a version spec containing $2, are locked to default with default value of false
+yq '.[] | select(.versionedSpecs[] | select(.version == "'$2'" and .lockToDefault == true and .default == false)) | .name' "${out_dir}/versioned_featuregates_${version}.yaml"
+yq '.[] | select(.versionedSpecs[] | select(.version == "'$2'" and .lockToDefault == true and .default == false)) | .name' "${out_dir}/unversioned_featuregates_${version}.yaml"
```

Comment:
Are these all the features gates locked to default true/false? Don't we need only the feature gates which are locked to default true/false compared to the old version?

Reply:
For reference, the old revision of the script was reporting only the locked to default feature gates compared to the old version. Ref https://github.com/gardener/gardener/blob/d4d2c39b9f50fc414210213885cf7d24d7fb4351/hack/compare-k8s-feature-gates.sh#L74-L76

Reply:
`select(.version == "'$2'" and .lockToDefault == true and .default == true)` selects feature gates that have a version spec containing the new version that has `lockedToDefault` set to true or false. Example:
```go
	TopologyAwareHints: {
		{Version: version.MustParse("1.21"), Default: false, PreRelease: featuregate.Alpha},
		{Version: version.MustParse("1.23"), Default: false, PreRelease: featuregate.Beta},
		{Version: version.MustParse("1.24"), Default: true, PreRelease: featuregate.Beta},
		{Version: version.MustParse("1.33"), Default: true, PreRelease: featuregate.GA, LockToDefault: true},
	},
```